### PR TITLE
[Movement] Restore Calibrations

### PR DESCRIPTION
**Pull request by maltewae**
Allows to restore calibrations that were saved on the hard disk. An example workflow is shown below:

Screenshot 2022-08-31 at 18 11 20
Screenshot 2022-08-31 at 18 11 24
Screenshot 2022-08-31 at 18 11 28
Screenshot 2022-08-31 at 18 11 36

### CODE REVIEW

File: LabExT/View/Movement/LoadStoredCalibrationWindow.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        chip: Type[Chip]
+    ) -> None:
+        """
+        Parameters
+        ----------
+        master : Tk
+            Tk instance of the master toplevel
+        mover : Mover
+            Instance of the current mover.
+        mover : Chip
```
Comment (marcoep) [inline LabExT/View/Movement/LoadStoredCalibrationWindow.py:40]: typo

File: LabExT/View/Movement/LoadStoredCalibrationWindow.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        ------
+        RuntimeError
+            If calibrations could not be restored.
+        """
+        self.mover: Type[MoverNew] = mover
+        self.chip: Type[Chip] = chip
+
+        if not messagebox.askyesno(
+            "Restore calibration",
+                f"Found mover calibration for chip: {self.chip.name}. Do you want to restore it?"):
```
Comment (marcoep) [inline LabExT/View/Movement/LoadStoredCalibrationWindow.py:53]: How do you know that we found a calibration at this point? There is no prior check if one is available, no?

File: LabExT/View/Movement/LoadStoredCalibrationWindow.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        self.calibration_options = [
+            f"Orientation: {c.get('orientation')} - Port: {c.get('device_port')}" for c in self.stored_calibrations]
+        self.calibration_options.append(self.ASSIGNMENT_MENU_PLACEHOLDER)
+
+        if self.calibration_settings.get("chip_name") != self.chip.name:
+            if not messagebox.askokcancel(
+                "Confirm restoring.",
+                f"The calibration file {self.mover.calibration_settings_file} was calibrated "
+                f"with chip {self.calibration_settings.get('chip_name')}. "
+                    f"You imported chip {self.chip.name}. Are you sure to continue?"):
```
Comment (marcoep) [inline LabExT/View/Movement/LoadStoredCalibrationWindow.py:79]: Suggestion: include the timestamp on when the calibration was created and saved. This helps the user to see if he needs to re-do it, e.g. when coming in again in the morning.

**Review by marcoep — CHANGES_REQUESTED** [review]

File: LabExT/View/Movement/LoadStoredCalibrationWindow.py (hunk 1 above)
Comment (maltewae) [reply LabExT/View/Movement/LoadStoredCalibrationWindow.py]: Done

File: LabExT/View/Movement/LoadStoredCalibrationWindow.py (hunk 2 above)
Comment (maltewae) [reply LabExT/View/Movement/LoadStoredCalibrationWindow.py]: There was a check before the window gets called. But you are right, here it is missing, I added one.

File: LabExT/View/Movement/LoadStoredCalibrationWindow.py (hunk 3 above)
Comment (maltewae) [reply LabExT/View/Movement/LoadStoredCalibrationWindow.py:91]: Added :)

File: LabExT/View/Movement/LoadStoredCalibrationWindow.py [hunk 4]
```diff
@@ -61,6 +57,22 @@ def __init__(
         with open(self.mover.calibration_settings_file, "r") as fp:
             self.calibration_settings = json.load(fp)
 
+        if self.calibration_settings.get("chip_name") != self.chip.name:
+            raise RuntimeError(
+                f"There is no stored calibrarion for the chip name '{self.chip.name}'."
+                f"Instead found one for chip name '{self.calibration_settings.get('chip_name')}'")
```
Comment (marcoep) [inline LabExT/View/Movement/LoadStoredCalibrationWindow.py:63]: I would only raise a warning here and not fail. Could be that the user did a typo? Force-failing on unequal chip names is a bit patronizing.

**Review by marcoep — CHANGES_REQUESTED** [review]

File: LabExT/View/Movement/LoadStoredCalibrationWindow.py (hunk 4 above)
Comment (maltewae) [reply LabExT/View/Movement/LoadStoredCalibrationWindow.py]: Changed :)

**Review by marcoep — APPROVED** [review]

**Pull request merged**